### PR TITLE
feat(api): community and moderators for project feed

### PR DESCRIPTION
**Pull request by Angshu31**
WARNING: Do not merge until this is tested

### CODE REVIEW

**Review by firecraftgaming — CHANGES_REQUESTED** [review]
seems good only problem right now is that the changes you did in JoinProjectArgs makes so that a user that joins can select their member role. I think its fine if you just remove the changes to that file

**Review by firecraftgaming — CHANGES_REQUESTED** [review]
and the changes to JoinProject is also not needed because a user that is joining shouldn't be able to ask for a role. You start as a member and then a staff team can change your role

File: api/prisma/schema.prisma [hunk 1]
```diff
@@ -164,6 +170,8 @@ model ProjectMember {
   roles            ProjectRole[]
   awaitingApproval Boolean
 
+  type ProjectMemberType
```
Comment (firecraftgaming) [inline api/prisma/schema.prisma:173]: edit this line to have at the end of it `@default(communityMember)`

**Review by firecraftgaming — CHANGES_REQUESTED** [review]
and add the default to type for member

**Review by ajhalili2006 — COMMENTED** [review]
Not probably a PR review, but for commit <https://github.com/hydralite/hydralite/pull/145/commits/b6f3f20517dfc7705810f07b7484f0659f8e2160>, please reword it to something like this...

```
chore(global): synchorize with upstream's dev branch
```

..so the Commitlint CI will go green.

**Comment by Angshu31** [comment]
While the code looks like it should work, and TS doesn't throw any errors, can @fullstackslayer still test this? If I do it before you do, I'll comment here to let you know it works.

**Comment by fullstackslayer** [comment]
> While the code looks like it should work, and TS doesn't throw any errors, can @fullstackslayer still test this? If I do it before you do, I'll comment here to let you know it works.

@Angshu31 Idk if I'll find time in the next 36 hours - I'm keeping busy with the notion doc and some personal stuff as well.

If I am able to check it out, I'll test it and inform you.

**Comment by Angshu31** [comment]
> > While the code looks like it should work, and TS doesn't throw any errors, can @fullstackslayer still test this? If I do it before you do, I'll comment here to let you know it works.
> 
> @Angshu31 Idk if I'll find time in the next 36 hours - I'm keeping busy with the notion doc and some personal stuff as well.
> 
> If I am able to check it out, I'll test it and inform you.

No problem. I might have tested it already by the time that you're ready to do it. I'll inform you if I do get around to testing it.

**Comment by fullstackslayer** [comment]
@Angshu31 what's the current status of this implementation
